core/domain/user_services_test.py

```diff
@@ -252,6 +252,68 @@ def test_set_and_get_user_email_preferences(self):
         self.assertEquals(
             email_preferences['can_receive_feedback_message_email'], False)
 
+    def test_set_and_get_user_email_preferences_for_exploration(self):
+        user_id = 'someUser'
+        exploration_id = 'someExploration'
+        username = 'username'
+        user_email = 'user@example.com'
+
+        user_services.get_or_create_user(user_id, user_email)
+        user_services.set_username(user_id, username)
+
+        # When ExplorationUserDataModel is yet to be created,
+        # the value of mute_feedback_notifications and
+        # mute_suggestion_notifications should be False.
+        email_preferences = user_services.get_email_preferences_for_exploration(
+            user_id, exploration_id)
+        self.assertEquals(
+            email_preferences['mute_feedback_notifications'],
+            feconf.DEFAULT_FEEDBACK_NOTIFICATIONS_MUTED_PREFERENCE)
+        self.assertEquals(
+            email_preferences['mute_suggestion_notifications'],
+            feconf.DEFAULT_SUGGESTION_NOTIFICATIONS_MUTED_PREFERENCE)
+
+        # This initializes a ExplorationUserDataModel instance with
```

Comment:
Before this, you could consider checking that the model doesn't exist in the datastore, so that the next step actually has an effect.